draft-pda-protocol.md

```diff
@@ -529,81 +527,79 @@ example, that clients are prohibited from talking to helpers but not the leader.
+`task` fields of the previous request. In addition, it includes the time (in
+seconds since the beginning of UNIX time) at which the report was generated, and
+the leader and helper share, both of which are structured as follows:
 
-[OPEN ISSUE: Is it safe to generate the proof once, then secret-share between
-each (leader, helper) pair? Probably not in general, but maybe for Prio?]
+~~~
+struct {
+  uint8 config_id;
+  opaque enc<1..2^16-1>;
+  opaque payload<1..2^16-1>;
+} PAEncryptedInputShare;
+~~~
 
-[OPEN ISSUE: allow server to send joint randomness in UploadStartResp, and then
-enforce uniqueness via double-spend state or something else (see issue#48).]
+* `config_id` is equal to `HpkeConfig.id`, where `HpkeConfig` is the
+  aggregator's key config.
+* `enc` is the encapsulated HPKE context, used by the aggregator to decrypt its
+  input share.
+* `payload` is the encrypted input share.
 
-The payload of the POST request to `[leader]/upload_finish` is structured as
-follows:
+To generate the report, the client begins by encoding its measurements as an
+input for the PA protocol and splitting it into a leader share `leader_share`
+and a helper share `helper_share`. (This procedure is specified by the PA
+protocol.) Each input share has the following format:
 
 ~~~
 struct {
-  PATask task;                 // Equal to PAUploadStartReq.task
-  uint64 time;                 // UNIX time (in seconds).
-  uint8 helper_hpke_config_id; // Equal to HpkeConfig.id
-  PAHelperInputShare helper_input_share;
-  PALeaderInputShare leader_input_share;
-} PAUploadFinishReq;
+  select (PAParam.proto) { // PAParam for the PA task
+    case prio: PrioInputShare;
+    case hits: hitsInputShare;
+  }
+} PAInputShare;
 ~~~
 
-We sometimes refer to this message as the *report*. The message contains the
-`task` fields of the previous request. In addition, it includes the time (in
-seconds since the beginning of UNIX time) at which the report was generated, the
-helper's HPKE config id, and the helper and leader shares. The helper share has
-the following structure:
+Next, the client generates an HPKE {{!I-D.irtf-cfrg-hpke}} context for the
+leader and helper by running
 
 ~~~
-struct {
-  opaque enc<1..2^16-1>;
-  PAProto proto;
-  select (PAHelperInputShare.proto) {
-    case prio: PrioHelperInputShare;
-    case hits: HitsHelperInputShare;
-  }
-} PAHelperInputShare;
+leader_enc, leader_context = SetupBaseS(leader_pk, [TODO])
+helper_enc, helper_context = SetupBaseS(helper_pk, [TODO])
 ~~~
 
-Field `enc` encodes the helper's encapsulated HPKE context. The remainder of the
-structure contains the share itself, the structure of which is specific to the
-PA protocol. The structure of the leader share is similarly protocol specific:
+`leader_pk` and `helper_pk` are, respectively, the leader and helper public key.
+`leader_enc` and `helper_enc` are the encapsulated HPKE context of each
+aggregator. Finally, the payloads of the encrypted input shares are computed as
```

Comment:
This is another opportunity to avoid duplication, but just talking about aggregators.

Reply:
Done.